### PR TITLE
Serve file index from a default point in the directory specified by t…

### PR DESCRIPTION
**Pull request by tpetersen0308**
…he user.

- The user must now specify a directory when starting the server.
- Also refactored renderWithHeaders() as overloaded render()
    in ResponseHelpers.
- Also refactored to test ResponseHelpers more directly, without
    stubbing out controller actions.

### CODE REVIEW

File: bin/ci/http_server [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -ex
+
+java -cp target/http-server-1.0-SNAPSHOT.jar server.Main 5000 /Users/Travis/Development/8th-Light/Apprenticeship/projects/resources/http-server/public
```
Comment (ssunday) [inline bin/ci/http_server:5]: couldn't this be relative so it doesn't have your dir info in it

Comment (tpetersen0308) [reply bin/ci/http_server:5]: I spent a little time trying to figure out how to get relative paths to work and couldn't crack it. I would prefer it though--I'll look into it some more.

Comment (ssunday) [reply bin/ci/http_server:5]: Yeah, cause right now it won't work for anyone but you...

what have you tried thus far? Could you change code to take relative and use file tools to find absolute?

Comment (tpetersen0308) [reply bin/ci/http_server:5]: Which would be why my CI builds are failing! I'm looking into that right now, thanks!

Comment (tpetersen0308) [reply bin/ci/http_server:5]: That did it! Thanks!

**Pull request merged**